independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/ModelUtils.java

```diff
@@ -184,6 +210,39 @@ public static String getVersion(Model model) {
         throw new IllegalStateException("Failed to determine version for project model");
     }
 
+    public static String getVersion(Model model) {
+        final String rawVersion = getRawVersion(model);
+        return isUnresolvedVersion(rawVersion) ? resolveVersion(rawVersion, model) : rawVersion;
+    }
+
+    public static boolean isUnresolvedVersion(String version) {
+        return version.charAt(0) == '$';//UNRESOLVED_VERSION_PATTERN.matcher(version).find();
```

Comment:
couldn't the property expression be in any part of the version string ?

Reply:
Since I was the one who added `UNRESOLVED_VERSION_PATTERN` I am also a litte surprised by this change.

The [Maven documentation](https://maven.apache.org/maven-ci-friendly.html) doesn't exactly answer this question but I don't see why you should _not_ be allowed to start the raw version with a literal.

It seems the only thing that is explicitly _forbidden_ as per above mentioned doc is the usage of any _other property than `revision`, `sha1` or `changelist`.

Reply:
I'll change it back.